aisynphys/connectivity.py

```diff
@@ -155,45 +158,21 @@ def measure_connectivity(pair_groups, alpha=0.05, sigma=None, fit_model=None, co
     for key, class_pairs in pair_groups.items():
         pre_class, post_class = key
         
-        probed_pairs = [p for p in class_pairs if pair_was_probed(p, pre_class.output_synapse_type)]
-        connections_found = [p for p in probed_pairs if p.has_synapse]
+        class_results = get_cp_results(class_pairs, alpha=alpha)
+        results[(pre_class, post_class)] = class_results
         
-        gaps_probed = [p for p in class_pairs if pair_probed_gj(p)]
-        gaps_found = [p for p in gaps_probed if p.has_electrical]
-
-        n_connected = len(connections_found)
-        n_probed = len(probed_pairs)
-        n_gaps_probed = len(gaps_probed)
-        n_gaps = len(gaps_found)
-        conf_interval_cp = connection_probability_ci(n_connected, n_probed, alpha=alpha)
-        conn_prob = float('nan') if n_probed == 0 else n_connected / n_probed
-        conf_interval_gap = connection_probability_ci(n_gaps, n_gaps_probed, alpha=alpha)
-        gap_prob = float('nan') if n_gaps_probed == 0 else n_gaps / n_gaps_probed
-
-        results[(pre_class, post_class)] = {
-            'n_probed': n_probed,
-            'n_connected': n_connected,
-            'n_gaps_probed': n_gaps_probed,
-            'n_gaps': n_gaps,
-            'connection_probability': (conn_prob,) + conf_interval_cp,
-            'gap_probability': (gap_prob,) + conf_interval_gap,
-            'connected_pairs': connections_found,
-            'gap_pairs': gaps_found,
-            'probed_pairs': probed_pairs,
-        }
-
-        distances = np.array([getattr(p, dist_measure) for p in probed_pairs], dtype=float)
-        connections = np.array([p.synapse for p in probed_pairs], dtype=bool)
-        gap_distances = np.array([getattr(p, dist_measure) for p in gaps_probed], dtype=float)
-        gaps = np.array([p.has_electrical for p in gaps_probed], dtype=bool)
+        distances = np.array([getattr(p, dist_measure) for p in class_results['probed_pairs']], dtype=float)
+        connections = np.array([p.synapse for p in class_results['probed_pairs']], dtype=bool)
+        gap_distances = np.array([getattr(p, dist_measure) for p in class_results['gaps_probed']], dtype=float)
+        gaps = np.array([p.has_electrical for p in class_results['gaps_probed']], dtype=bool)
         mask = np.isfinite(distances) & np.isfinite(connections)
         results[(pre_class, post_class)]['probed_distances'] = distances[mask]
         results[(pre_class, post_class)]['connected_distances'] = connections[mask]
         mask2 = np.isfinite(gap_distances) & np.isfinite(gaps)
         results[(pre_class, post_class)]['gap_probed_distances'] = gap_distances[mask2]
         results[(pre_class, post_class)]['gap_distances'] = gaps[mask2]
         
-        if sigma is not None and fit_model is not None:
```

Comment:
This got changed from an `or` to an `and` in this [commit](https://github.com/AllenInstitute/aisynphys/commit/6fe292a54d0e2b0b82589e93903904df5920e287). I think we could actually just ditch the sigma check. Making it an `and` prevents current use in some of the notebooks.